hack/local-up-cluster.sh

```diff
@@ -1088,6 +1088,11 @@ fi
 
 if [ "${CONTAINER_RUNTIME}" == "docker" ] && ! kube::util::ensure_docker_daemon_connectivity; then
   exit 1
+else
+  # docker doesn't allow to reach exposed hostPorts from the node, however, Kubernetes does
+  # so we append a new rule on top of the docker one
+  # -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER  <-- docker rule
+  iptables -t nat -A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
```

Comment:
the gce jobs use a different approach not allowing docker to manage the iptables rules
https://github.com/kubernetes/kubernetes/blob/2eb6911e832152abb1cbfd31ae15ceceb4e844a0/cluster/gce/gci/configure-helper.sh#L1515-L1517 and uses an userland proxy to forward the ports

Reply:
right, the trade off is we will need a restart if we update the daemon.json

Reply:
just wanted to document it, I was trying to understand how it was working in gce and not here.
Pick your poison XD

Reply:
thanks for tracking this down @aojea ! huge help.

Reply:
shouldn't we only do this if `"${CONTAINER_RUNTIME}" == "docker"`?

Reply:
good point, fixing

Reply:
unrelated to the PR but does the script supports other runtimes?
I've tried to use containerd and didn't work and I couldn't see the same code used for the gce setup to allow others.

Reply:
@aojea i use this script with containerd regularly!

Reply:
Good to know, a new tool for my box. Maybe I made some mistake